kolibri/utils/server.py

```diff
@@ -356,14 +356,16 @@ def get_status():  # noqa: max-complexity=16
             # Probably a mis-configured kolibri
             raise NotRunning(STATUS_SERVER_CONFIGURATION_ERROR)
 
-        return pid, LISTEN_ADDRESS, listen_port  # Correct PID !
-
     else:
         try:
             requests.get(check_url, timeout=3)
+        except (requests.exceptions.ReadTimeout,
+                requests.exceptions.ConnectionError):
+            raise NotRunning(STATUS_NOT_RESPONDING)
         except (requests.exceptions.RequestException):
             return pid, '', ''
 
+    return pid, LISTEN_ADDRESS, listen_port  # Correct PID !
```

Comment:
The way this entire function is set up is a bit confusing and I'm worried it might be brittle with all the if/else/try/except/return/raise branches.

Something to look at refactoring later, maybe by breaking it up into separate functions